DigitalLearningSolutions.Data.Tests/DataServices/UserDataServiceTests/UserCentreDetailsDataServiceTests.cs

```diff
@@ -57,5 +60,32 @@ public void EmailIsInUseByOtherUser_returns_true_if_email_is_in_use_by_other_use
             // Then
             result.Should().Be(expectedResult);
         }
+
+        [Test]
+        [TestCase(true, null)]
+        [TestCase(true, "new@admin.email")]
+        [TestCase(false, null)]
+        [TestCase(false, "new@admin.email")]
+        public void SetCentreEmail_sets_email_if_not_empty(bool detailsExist, string? email)
+        {
+            using var transaction = new TransactionScope();
+
+            // Given
+            if (detailsExist)
+            {
+                connection.Execute(
+                    @"INSERT INTO UserCentreDetails (UserID, CentreID, Email)
+                        VALUES (8, 374, 'sample@admin.email')"
+                );
+            }
+
+            // When
+            userDataService.SetCentreEmail(8, 374, email);
+            var result = connection.Query<string?>(@"SELECT Email FROM UserCentreDetails WHERE UserID = 8")
+                .SingleOrDefault();
+
+            // Then
+            result.Should().BeEquivalentTo(email);

```

Comment:
It would be good to have some way of distinguishing between null because the record doesn't exist and null because the record contains a null if possible so that we can confirm the "false, null" case doesn't create a record . If not, I guess this is enough